### ISSUE TITLE
Issue passing data to Linux shell subprocesses

### ISSUE THREAD

**Issue by sjaenick** [dotnet/runtime#49203]
### Description

Starting an external subprocess via Process/ProcessStartInfo with stdin/stdout redirected, 
data can be successfully passed to the processes stdin as long as it is a binary being
executed. For a shell script, however, this fails and the subprocess is unable to read data
from /dev/fd/0.

### Configuration
```
$ dotnet --info
.NET SDK (reflecting any global.json):
 Version:   5.0.200
 Commit:    70b3e65d53

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /vol/mgx-sw/lib/dotnet/sdk/5.0.200/

Host (useful for support):
  Version: 5.0.3
  Commit:  c636bbdc8a

.NET SDKs installed:
  5.0.200 [/vol/mgx-sw/lib/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.3 [/vol/mgx-sw/lib/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.3 [/vol/mgx-sw/lib/dotnet/shared/Microsoft.NETCore.App]
```

Output of testcase
```
$ ./PipeTest
/homes/sjaenick/bin/test-cat.sh: 3: cannot open /dev/fd/0: No such device or address
Unhandled exception. System.IO.IOException: Unable to write data to the transport connection: Broken pipe.
 ---> System.Net.Sockets.SocketException (32): Broken pipe
   at System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   --- End of inner exception stack trace ---
   at System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.StreamWriter.WriteLine(String value)
   at Pipe.PipeTest.Main(String[] args) in /vol/mgx-sw/src/tools/conveyor/Contrib/PipeTest/PipeTest.cs:line 30
Aborted
```

PipeTest.cs
```
using System;
using System.Diagnostics;
using System.IO;
using System.Text;

namespace Pipe
{
    public class PipeTest
    {
        public PipeTest()
        {
        }

        static void Main(string[] args) {

            // this works
            //ProcessStartInfo psi = new ProcessStartInfo("/usr/bin/cat");
           
            // this doesn't
            ProcessStartInfo psi = new ProcessStartInfo("/homes/sjaenick/bin/test-cat.sh");
            psi.RedirectStandardInput = true;
            psi.RedirectStandardOutput = true;
            psi.UseShellExecute = false;
            psi.CreateNoWindow = true;

            System.Diagnostics.Process proc = new System.Diagnostics.Process();
            proc.StartInfo = psi;
            proc.Start();

            proc.StandardInput.WriteLine("teststring");
            proc.StandardInput.Close();

            using (StreamReader sr = proc.StandardOutput) {
                string line;
                while ((line = sr.ReadLine()) != null)
                {
                    System.Console.WriteLine(line);
                }
            }
        }
    }
}

```

test-cat.sh
```
#!/bin/sh

cat < /dev/fd/0
```

### Regression?

Don't know.

**Comment by stephentoub** [comment]
Looks like a dup of https://github.com/dotnet/runtime/issues/46469 which has been fixed for 5.0.4.

**Comment by sjaenick** [comment]
> Looks like a dup of #46469 which has been fixed for 5.0.4.

Wasn't able to find a release schedule - what's the ETA for 5.0.4?

**Comment by stephentoub** [comment]
I expect in the next week or two.

**Comment by sjaenick** [comment]
Issue is fixed in 5.0.4, closing..
